pkg/splunk/enterprise/indexercluster.go

```diff
@@ -118,23 +122,168 @@ func ApplyIndexerCluster(client splcommon.ControllerClient, cr *enterprisev1.Ind
 
 	// no need to requeue if everything is ready
 	if cr.Status.Phase == splcommon.PhaseReady {
+		if len(cr.Status.IndexerSecretChanged) > 0 {
+			// Disable maintenance mode
+			err = SetCmMaintenanceMode(client, cr, false, false)
+			if err != nil {
+				return result, err
+			}
+		}
+
+		// Reset idxc secret changed and namespace secret revision
+		cr.Status.IndexerSecretChanged = []bool{}
+		cr.Status.NamespaceSecretResourceVersion = namespaceScopedSecret.ObjectMeta.ResourceVersion
+
 		result.Requeue = false
 	}
 	return result, nil
 }
 
 // indexerClusterPodManager is used to manage the pods within an indexer cluster
 type indexerClusterPodManager struct {
+	c               splcommon.ControllerClient
 	log             logr.Logger
 	cr              *enterprisev1.IndexerCluster
 	secrets         *corev1.Secret
 	newSplunkClient func(managementURI, username, password string) *splclient.SplunkClient
 }
 
+// SetCmMaintenanceMode enables/disables cluster maintenance mode
+func SetCmMaintenanceMode(c splcommon.ControllerClient, cr *enterprisev1.IndexerCluster, enable bool, mock bool) error {
+	// Retreive admin password from Pod
+	masterIdxcName := cr.GetName()
+	// For multisite / multipart cluster, use the cluster-master API of the referenced IndexerCluster
+	if len(cr.Spec.ClusterMasterRef.Name) > 0 {
+		masterIdxcName = cr.Spec.ClusterMasterRef.Name
```

Comment:
I think ClusterMasterRef is not an optional in this case, and it should be there, or else we need to return?

Reply:
Taken care of